controller/editorcontroller.php

```diff
@@ -904,9 +916,30 @@ public function config($fileId, $filePath = null, $shareToken = null, $version =
             }
         }
 
+        $isPersistentLock = false;
+        if ($version < 1
+            && (\OC::$server->getConfig()->getAppValue("files", "enable_lock_file_action", "no") === "yes")) {
+
+            $storageId = $fileStorage->getStorageCache()->getNumericId();
+            $targetUserId = !empty($userId) ? $userId : $file->getOwner()->getUid();
+            $userFolder = $this->root->get($targetUserId);
+            $path = substr($userFolder->getRelativePath($file->getPath()), 1);
```

Comment:
Building the path like this could cause some issues in specific scenarios. When User A shares a file with User B, it is possible that the file has a different name under User B. This usually happens when User B already has a file which has the exact name like the shared one. This implementation would check for the "new" filename (the one for User B). The database has the lock under its original filename though, so no locks can be found.

Better let the storage take care of `getLocks()`:

```
if ($fileStorage->instanceOfStorage(IPersistentLockingStorage::class)) {
    $locks = $fileStorage->getLocks($file->getFileInfo()->getInternalPath(), false);
    ...
}
```